File: 백준/Silver/1931. 회의실 배정/회의실 배정.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        int N = Integer.parseInt(br.readLine());
        int[][] arr = new int[N][2];
        int answer = 0;
        int before = -1;

        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            arr[i][0] = Integer.parseInt(st.nextToken());
            arr[i][1] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(arr, (a, b) -> a[1] == b[1] ? a[0] - b[0] : a[1] - b[1]);

        for (int i = 0; i < N; i++) {
            if (before <= arr[i][0]) {
                before = arr[i][1];
                answer++;
            }
        }

        System.out.println(answer);
    }
}

File: 백준/Silver/1931. 회의실 배정/회의실 배정.kt
import java.util.*

fun main() {
    data class Meeting(val start: Int, val end: Int)

    val n = readLine()!!.toInt()
    val meetings = mutableListOf<Meeting>()
    repeat(n) {
        val st = StringTokenizer(readLine())
        meetings.add(Meeting(st.nextToken().toInt(), st.nextToken().toInt()))
    }
    meetings.sortWith(compareBy({ it.end }, { it.start }))
    var count = 1
    var current = meetings[0].end
    for (i in 1 until n) {
        if (meetings[i].start >= current) {
            count++
            current = meetings[i].end
        }
    }
    print(count)
}
